src/router/parametersPreProcess.routerPlugin.js

```diff
@@ -0,0 +1,86 @@
+import { MAP_VIEWS } from '@/router/viewNames'
+import { WEBMERCATOR } from '@/utils/coordinates/coordinateSystems'
+import log from '@/utils/logging'
+import { isNumber } from '@/utils/numberUtils'
+
+/**
+ * @param {String} param_name
+ * @param {any} param_value
+ * @param {Object} newQuery The modified query for the router
+ */
+const preProcessParam = (param_name, param_value, query) => {
+    switch (param_name) {
+        case '3d':
+            // if the 3d flag is set, we need to ensure that we are in the webmercator projection
+            if (
+                ((typeof param_value === 'boolean' && param_value) ||
+                    param_value === 'true' ||
+                    param_value === null) &&
+                (!isNumber(query.sr) || Number(query.sr) !== WEBMERCATOR.epsgNumber)
```

Comment:
I would keep like before, to keep it simple, and always overwrite the sr parameter even if it has already the correct value